python/lsst/ts/externalscripts/base_make_calibrations.py

```diff
@@ -0,0 +1,640 @@
+            image_types.append("PTC")
+
+        exposure_ids_dict = {"BIAS": (), "DARK": (), "FLAT": ()}
+        for im_type in image_types:
+            # 1. Take images with the instrument, only for "BIAS,
+            # "DARK", or "FLAT".
+            if im_type in ["BIAS", "DARK", "FLAT"]:
+                if checkpoint:
+                    if im_type == "BIAS":
+                        await self.checkpoint(f"Taking {self.config.n_bias} biases.")
+                    elif im_type == "DARK":
+                        await self.checkpoint(f"Taking {self.config.n_dark} darks.")
+                    elif im_type == "FLAT":
+                        await self.checkpoint(f"Taking {self.config.n_flat} flats.")
+
+                # TODO: Before taking flats with LATISS (and also
+                # with LSSTComCam), check that the telescope is in
+                # position to do so. See DM-31496, DM-31497.
+                exposure_ids = await self.take_images(im_type)
+                exposure_ids_dict[im_type] = exposure_ids
+
+                if checkpoint:
+                    # Image IDs
+                    await self.checkpoint(f"Images taken: {exposure_ids}; type: {im_type}")
+
+            # 2. Call the calibration pipetask via the OCPS to make a master
+            response_ocps_calib_pipetask = await self.call_pipetask(im_type, exposure_ids_dict)
+
+            # 3. Verify the calibration (implemented so far for bias,
+            # dark, and flat).
+            job_id_calib = response_ocps_calib_pipetask['jobId']
+            if self.config.do_verify and im_type in ["BIAS", "DARK", "FLAT"]:
+                if not response_ocps_calib_pipetask['phase'] == 'completed':
+                    raise RuntimeError(f"{im_type} generation not completed successfully: "
+                                       f"Status: {response_ocps_calib_pipetask['phase']}. "
+                                       f"{im_type} verification could not be performed.")
+                else:
+                    response_ocps_verify_pipetask = await self.verify_calib(im_type,
+                                                                            job_id_calib, exposure_ids_dict)
+                    previous_step = "verification"
+            else:
+                self.log.info(f"Skipping verification for {im_type}. ")
+                response_ocps_verify_pipetask = response_ocps_calib_pipetask
+                previous_step = "generation"
+
+            # 4. Certify the calibration
+            if not response_ocps_verify_pipetask['phase'] == 'completed':
+                raise RuntimeError(f"{im_type} {previous_step} not completed successfully: "
+                                   f"Status: {response_ocps_verify_pipetask['phase']}. "
+                                   f"{im_type} certification could not be performed.")
+            else:
+                await self.certify_calib(im_type, job_id_calib)
```

Comment:
This will add a check when we move out of auto-certify mode, correct?  Should that ticket be filed and noted in a comment here so that that doesn't get forgotten?

Reply:
Would that be ticket DM-31897?

Reply:
I added a reference to that ticket in a comment.